src/contracts/core/AllocationManager.sol

```diff
@@ -55,138 +63,131 @@ contract AllocationManager is
     ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
         require(0 < params.wadToSlash && params.wadToSlash <= WAD, InvalidWadToSlash());
 
-        // Check that the operator is registered and slashable
-        OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: params.operatorSetId});
-        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
-        require(avsDirectory.isOperatorSlashable(params.operator, operatorSet), InvalidOperator());
-
-        // Record the proportion of 1e18 that the operator's total shares that are being slashed
-        uint256[] memory wadSlashed = new uint256[](params.strategies.length);
-
-        for (uint256 i = 0; i < params.strategies.length; ++i) {
-            PendingMagnitudeInfo memory info =
-                _getPendingMagnitudeInfo(params.operator, params.strategies[i], operatorSetKey);
+        // Check that the operator set exists and the operator is slashable for the set
+        OperatorSet memory operatorSet = OperatorSet(msg.sender, params.operatorSetId);
+        require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
+        require(isOperatorSlashable(params.operator, operatorSet), InvalidOperator());
+
+        uint256 length = _operatorSetStrategies[operatorSet.key()].length();
+        IStrategy[] memory strategiesSlashed = new IStrategy[](length);
+        uint256[] memory wadSlashed = new uint256[](length);
+
+        // For each strategy in the operator set, slash any existing allocation
+        for (uint256 i = 0; i < length; i++) {
+            // 1. Get the operator's allocation info for the strategy and operator set
+            IStrategy strategy = IStrategy(_operatorSetStrategies[operatorSet.key()].at(i));
+            (StrategyInfo memory info, Allocation memory allocation) =
+                _getUpdatedAllocation(msg.sender, operatorSet.key(), strategy);
+            strategiesSlashed[i] = strategy;
+
+            // 2. Skip if the operator is not allocated to the operator set from this strategy
+            if (allocation.currentMagnitude == 0) {
```

Comment:
IMO should still have a hard requirement that `currentMagnitude > 0`. Offchain systems having to process a slashing where currentMag == 0 doesn't make sense

Reply:
we can't have this requirement, because an operator may not have a strategy allocated! this would create unslashable operators (unless im misunderstanding)

Reply:
one thing we could _maybe_ do is have an enumerable set of strategies that have existing allocations:

```
mapping(operator => operatorSetKey => EnumerableSet.AddressSet) allocationList;
```

... and we could iterate over this list when slashing.

Reply:
Actually i dont think my solution is possible - we might end up removing allocatedStrategies in the middle of iterating over it, and the `.remove` method swaps indices around, so... dangerous!

Reply:
ok i have a solution here -- we iterate in reverse order.

and in fact i think my solution is a requirement, because we need to be able to fetch allocations that aren't necessarily recognized by an operator set. currently we lack the ability to do this, so we need an `allocationList` type tracker for this.

Reply:
> we can't have this requirement, because an operator may not have a strategy allocated

Ah true, I forgot that we have are not passing in a list of strategies. This is fine, just need offchain systems to properly handle the 0 slash case.

Reply:
Interesting. Yes, we definitely do want a tracker for this because given the below discussion it sounds like we aren't going to go and clear all _pending_ modifications. 

Another edge case is when a strategy is removed from the operatorSet, operator is still allocated, strategy is re-added. The operator is instantly allocatable

Reply:
[faf9a7b](https://github.com/Layr-Labs/eigenlayer-contracts/pull/860/commits/faf9a7b23e157652f16db8fee8c2f826b1aa1377)

added this enumeration here